### DISCUSSION TITLE
Potential funding to advance `image`

### DISCUSSION THREAD

**Discussion by Shnatsel** [image-rs/image#2288] [General]
I have started an (as yet unannonced) project to build a drop-in replacement for `imagemagick` in Rust: https://github.com/Shnatsel/wondermagick

The early benchmarks are really promising, with `hyperfine` reporting ~1.6x speedups over `imagemagick` on a number of common operations.

With these kinds of numbers I am hoping to get more attention onto the Rust imaging ecosystem, and try to secure funding to advance both `wondermagick` and `image`.

The kind of tasks that would need to be done on the `image` side may range from small things like https://github.com/image-rs/image-png/pull/488 to bigger work items like optimizing the WebP decoder, as well as general maintenance work.

Obviously I would like the maintainers of `image` to handle the work related to `image` and receive the funding for it. Would you be open to that?

In concrete terms, I could possibly get up to $50k (but easily could be less) with the work expected to be carried out over the course of several months. Depending on the source of funding, it might commence as soon as next month or as late as ~6 months from now.

If you are interested, how can I contact you privately to discuss the details? If we'll be working together, a lower-latency channel like a chat could be nice.

**Comment by Shnatsel** [comment]
If you cannot dedicate much of your own time to this, I could get someone else to do the brunt of the work, and compensate you for the code review.

@fintelia @HeroicKatora what do you think?

**Reply by 197g** [reply]
Interesting coincidence. I'm be open to other work (up to full time) starting mid of November. Without committing to _this_ immediately, seems clear enough to factor it in as a day per week or similar arrangements.

**Reply by Shnatsel** [reply]
Great! I have emailed a draft of the application to you and @fintelia. Please review it and let me know what you think!

I've looked at the scope of the work and I'll actually be applying for €100k because there's a lot to be done. I've also talked to @etemesi254 (author of `zune-jpeg`) and he will most likely be available to work on this full-time, if needed.
